lib/linked_list.rb

```diff
@@ -18,63 +18,146 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(value)
-      raise NotImplementedError
+      # raise NotImplementedError
+      
+        new_node = Node.new(value)
+        new_node.next = @head
+        @head = new_node
+  
+
     end
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def search(value)

```

Comment:
👍 , however the space complexity is O(1) because you're not creating new nodes here.